### PR TITLE
Add support for new custom contract

### PR DESCRIPTION
**Pull request by adam-maj**

### CODE REVIEW

File: docs/common/custom.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ For example, if I had an NFT custom contract and I wanted to call the `balance`
 
 ```python
 # Any arguments passed to the balance function go into the args of "call"
-custom.functions.balance(1).call()
+contract.functions.balance(1).call()
```
Comment (joaquim-verges) [inline docs/common/custom.md:37]: Need to update this example and the one below right?

File: thirdweb/contracts/custom.py [hunk 2]
```diff
@@ -100,21 +102,56 @@ def __init__(
         self.nft = self._detect_erc_721()
         self.edition = self._detect_erc_1155()
 
-        # self.signature_mint = self.detect_signature_mint()
+    def call(self, fn: str, *args) -> Any:
```
Comment (joaquim-verges) [inline thirdweb/contracts/custom.py:105]: Nice. Add a doc + example here too ?

File: thirdweb/core/classes/ipfs_storage.py [hunk 3]
```diff
@@ -44,16 +44,22 @@ class IpfsStorage(ABC):
...
         """
 
         res = self._get(hash)
-        data = res.json()
+
+        # If the data is not JSON, return it raw
+        try:
+            data = res.json()
+        except Exception:
+            return data
```
Comment (joaquim-verges) [inline thirdweb/core/classes/ipfs_storage.py:61]: This seems not related to custom contract? What was the motivation here. Also we need to be careful because in most languages, you can only consume the response once. So just trying to get the json out of it makes the response unusable after that

**Review by joaquim-verges — COMMENTED** [review]
Looks really good, just a few comments

File: thirdweb/core/classes/ipfs_storage.py (hunk 3 above)
Comment (adam-maj) [reply thirdweb/core/classes/ipfs_storage.py:61]: Hmm probably better to leave it as it was before then for now - we can handle this properly in another PR, since we shouldn't just assume that the data is JSON (same goes for TS/Go SDK)

File: docs/common/custom.md (hunk 1 above)
Comment (adam-maj) [reply docs/common/custom.md:37]: Ah yes was editing the doc but hadn't pushed up 😄

File: thirdweb/contracts/custom.py (hunk 2 above)
Comment (adam-maj) [reply thirdweb/contracts/custom.py:105]: This class is internal for now on python - that's why the docs are manual in the other place, reason being that I prefer to have a handwritten doc for something as important as custom contract + people don't really need to see the generic custom contract type externally.

Comment (adam-maj) [reply thirdweb/contracts/custom.py:105]: We could def expose it though if necessary

Comment (joaquim-verges) [reply thirdweb/contracts/custom.py:105]: but the `call` function itself needs some doc no? why would this class be internal?
